experimental/web_dashboard/lib/src/experiments/routing_demo.dart

```diff
@@ -0,0 +1,205 @@
+// Copyright 2020, the Flutter project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter/scheduler.dart';
+
+import '../widgets/third_party/adaptive_scaffold.dart';
+
+/// The [Key] for the inner [Navigator].
+GlobalKey _navigatorKey = GlobalKey();
+
+void main() {
+  runApp(RoutingDemoApp());
+}
+
+/// A responsive dashboard using AdaptiveScaffold and NavigationRail with named
+/// routes. Routing works as expected using a few tricks.
+class RoutingDemoApp extends StatefulWidget {
+  @override
+  _RoutingDemoAppState createState() => _RoutingDemoAppState();
+}
+
+class _RoutingDemoAppState extends State<RoutingDemoApp> {
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      // Forward the initial route to the app shell.
+      onGenerateInitialRoutes: (initialRoute) {
+        return [
+          PageRouteBuilder(
+            // Use the same route settings in the child navigator.
+            settings: RouteSettings(name: initialRoute),
+            // Send the route to AppShell so it can select the correct item in
+            // the Drawer / NavigationRail / BottomAppBar.
+            pageBuilder: (context, animation, secondaryAnimation) {
+              return AppShell(initialRoute);
+            },
+          ),
+        ];
+      },
+    );
+  }
+}
+
+/// The outer "shell" of the app. Uses an [AdaptiveScaffold] with a [Navigator]
+/// to handle route events and display the correct page in the Drawer /
+/// BottomNavigationBar, or NavigationRail, as well as the correct page in the
+/// body of the Scaffold.
+class AppShell extends StatefulWidget {
+  final String routeName;
```

Comment:
Is this used solely for the initial route when building the navigator later? If so, consider renaming it to `initialRouteName` or something similar.

Reply:
considering this fixed since I took this file out.